### PR TITLE
Add support for mongo based druid lookups

### PR DESCRIPTION
**Pull request by patelh**
Address #417 by adding support for MongoExtractionNamespace.

Changes
* Rename InMemory to RocksDB
* Remove need to serialize and deserialize MahaLookupQueryElement for doing lookups with MahaLookupExtractionFn.
* Remove unused ObjectMapper from MahaRegisteredLookupExtractionFn
* Add tests for MahaNamespaceExtractionModule, MahaLookupExtractionFactory
* Add jacoco for code coverage
* Add assembly plugin to create zip file with dependent libs
* Add documentation on usage

### CODE REVIEW

**Comment by patelh** [comment]
@pavanab4u please take a look, still need to add more doc and test on local cluster

**Comment by patelh** [comment]
@pavanab4u I was able to successfully test MongoExtractionNamespace with a druid 0.11 cluster.

**Comment by patelh** [comment]
@pavanab4u can you please take a look at this, I am blocked until this is merged.

**Comment by pavanab4u** [comment]
@patelh I'll review today

File: druid-lookups/src/main/java/com/yahoo/maha/maha_druid_lookups/query/lookup/namespace/ExtractionNamespace.java [hunk 1]
```diff
@@ -4,16 +4,21 @@
 
 import com.fasterxml.jackson.annotation.JsonSubTypes;
 import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import com.google.common.collect.ImmutableList;
 
 @JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = "type")
 @JsonSubTypes(value = {
         @JsonSubTypes.Type(name = "mahajdbc", value = JDBCExtractionNamespace.class),
-        @JsonSubTypes.Type(name = "mahainmemorydb", value = InMemoryDBExtractionNamespace.class)
+        @JsonSubTypes.Type(name = "maharocksdb", value = RocksDBExtractionNamespace.class),
```
Comment (pavanab4u) [inline druid-lookups/src/main/java/com/yahoo/maha/maha_druid_lookups/query/lookup/namespace/ExtractionNamespace.java]: this rename is backward incompatible change.

File: druid-lookups/src/main/java/com/yahoo/maha/maha_druid_lookups/query/lookup/namespace/RocksDBExtractionNamespace.java [hunk 2]
```diff
@@ -13,8 +13,8 @@
 import javax.validation.constraints.NotNull;
 import java.util.Objects;
 
-@JsonTypeName("mahainmemorydb")
```
Comment (pavanab4u) [inline druid-lookups/src/main/java/com/yahoo/maha/maha_druid_lookups/query/lookup/namespace/RocksDBExtractionNamespace.java]: this rename is backward incompatible change.

Comment (patelh) [reply druid-lookups/src/main/java/com/yahoo/maha/maha_druid_lookups/query/lookup/namespace/RocksDBExtractionNamespace.java]: I believe this is only used for serialization.

File: druid-lookups/src/main/java/com/yahoo/maha/maha_druid_lookups/query/lookup/namespace/ExtractionNamespace.java (hunk 1 above)
Comment (patelh) [reply druid-lookups/src/main/java/com/yahoo/maha/maha_druid_lookups/query/lookup/namespace/ExtractionNamespace.java]: Yep, you are right, good catch.  I'll add a subtype for "mahainmemorydb".

**Pull request merged**